### PR TITLE
8310170: Use sp's argument to improve performance of outputStream::indent and remove SP_USE_TABS

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi,

Please consider this small enhancement. `sp` takes a count argument, this was never used by `indent`, let's just use it.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8310170](https://bugs.openjdk.org/browse/JDK-8310170): Use sp's argument to improve performance of outputStream::indent and remove SP_USE_TABS (**Enhancement** - P4)

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to [d9ee4c2f](https://git.openjdk.org/jdk/pull/14502/files/d9ee4c2f8093848b1b62b4890fa90fdd2530d4c3)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14502/head:pull/14502` \
`$ git checkout pull/14502`

Update a local copy of the PR: \
`$ git checkout pull/14502` \
`$ git pull https://git.openjdk.org/jdk.git pull/14502/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14502`

View PR using the GUI difftool: \
`$ git pr show -t 14502`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14502.diff">https://git.openjdk.org/jdk/pull/14502.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14502#issuecomment-1594333513)

### CODE REVIEW

**Review by shipilev — APPROVED** [review]
Looks okay. I thought what would happen if `_indentation < _position`, but `sp()` seems to handle `count < 0` fine.

**Review by dholmes-ora — APPROVED** [review]
There's potentially a subtle difference in behaviour here if SP_USE_TABS were ever true - but as it is defined to be false there is no difference in practice. Maybe we should just cull the code related to SP_USE_TABS and not pretend we could do something different here?

Thanks.

**Comment by jdksjolen** [comment]
Hi,

Sorry for the slow response from me on this.

@dholmes-ora, let's remove SP_USE_TABS in this PR. I'll change the ticket description.

File: src/hotspot/share/utilities/ostream.cpp [hunk 1]
```diff
@@ -186,18 +186,9 @@ void outputStream::put(char ch) {
   write(buf, 1);
 }
 
-#define SP_USE_TABS false
-
 void outputStream::sp(int count) {
   if (count < 0)  return;
```
Comment (tstuefe) [inline src/hotspot/share/utilities/ostream.cpp:190]: Is this  needed?

**Review by tstuefe — APPROVED** [review]
Good.

File: src/hotspot/share/utilities/ostream.cpp (hunk 1 above)
Comment (jdksjolen) [reply src/hotspot/share/utilities/ostream.cpp:190]: The `count < 0` check? Yes, as Shipilev mentioned:

>Looks okay. I thought what would happen if _indentation < _position, but sp() seems to handle count < 0 fine.

So it's nice that we don't have to care at the call site.

Comment (tstuefe) [reply src/hotspot/share/utilities/ostream.cpp:190]: Ah, okay. Missed that.

**Review by dholmes-ora — APPROVED** [review]
Looks good.

As an aside I was disappointed to discover that the indentation facility is far less useful than it might be. I had hoped you could set the indentation and then have all following log output indented, until you reset it. But indentation only happens when you explicitly call `indent()` so you can't call e.g. `val->print_on(stream)` and have the caller control indentation (without `print_on` directly supporting it).

**Comment by jdksjolen** [comment]
One structured concurrency test fails on GHA on Linux-x86. That can't be thanks to this patch. Integrating.

/integrate

**Pull request closed without merging**